### PR TITLE
WL-4661 Enhancements to QuickLinks.

### PR DESCRIPTION
**Pull request by RebeccaMiller-Which**
QuickLinks can now be per skin, also removed unused portalLinks code.

### CODE REVIEW

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java [hunk 1]
```diff
@@ -576,7 +576,16 @@ else if (twoFactorAuthentication.isTwoFactorRequired("/site/"+siteId)
 		rcontext.put("currentUrlPath", Web.serverUrl(req) + req.getContextPath()
 				+ URLUtils.getSafePathInfo(req));
 
-		doSendResponse(rcontext, res, null);
+        // Find any related links ready for display in the top navigation bar,
```
Comment (buckett) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java]: Indentation looks off.

File: portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java [hunk 2]
```diff
@@ -541,23 +536,18 @@ public void setSiteNeighbourhoodService(SiteNeighbourhoodService siteNeighbourho
...
-	
-	
+
 	public ContentHostingService getContentHostingService() {
 		return contentHostingService;
 	}
+
 	/**
 	 * @param portalLinks the portal icons to set
+     * @superseded by relatedLinks functionality also in this class.
```
Comment (buckett) [inline portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java]: Indentation.

File: portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java [hunk 3]
```diff
@@ -631,5 +621,82 @@ public String getSkinPrefix() {
...
+        // Find the related links title if it is in the properties file.
+		return serverConfigurationService.getString("related.link.info", "");
+	}
+
+	public List<Map> getRelatedLinks(String siteSkin){
+		/* Find the related links (if they are in the properties file) ready for display in the top navigation bar.
+		 * First try with the skin name as there may be different related links per site, then try with no skin. */
+		List<String> linkUrls = null;
+		List<String> linkTitles = null;
+		List<String>linkNames = null;
```
Comment (buckett) [inline portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java:634]: Formatting.

**Comment by RebeccaMiller-Which** [comment]
Hopefully all formatting issues resolved.

**Pull request merged**